.github/workflows/Build Archives Thunder on Linux.yml

```diff
@@ -0,0 +1,118 @@
+      with:
+        path: Thunder
+        repository: rdkcentral/Thunder
+
+    - name: Regex Thunder
+      if: contains(github.event.pull_request.body, '[DependsOn=Thunder:')
+      id: thunder
+      uses: AsasInnab/regex-action@v1
+      with:
+        regex_pattern: '(?<=\[DependsOn=Thunder:).*(?=\])'
+        regex_flags: 'gim'
+        search_string: ${{github.event.pull_request.body}}
+
+    - name: Checkout Thunder - ${{steps.thunder.outputs.first_match}}
+      if: contains(github.event.pull_request.body, '[DependsOn=Thunder:')
+      uses: actions/checkout@v3
+      with:
+        path: Thunder
+        repository: rdkcentral/Thunder
+        ref: ${{steps.thunder.outputs.first_match}}
+
+# ----- ThunderTools -----
+    - name: Checkout ThunderTools - default
+      if: ${{ !contains(github.event.pull_request.body, '[DependsOn=ThunderTools:') }}
+      uses: actions/checkout@v3
+      with:
+        path: ThunderTools
+        repository: rdkcentral/ThunderTools
+
+    - name: Regex ThunderTools
+      if: contains(github.event.pull_request.body, '[DependsOn=ThunderTools:')
+      id: tools
+      uses: AsasInnab/regex-action@v1
+      with:
+        regex_pattern: '(?<=\[DependsOn=ThunderTools:).*(?=\])'
+        regex_flags: 'gim'
+        search_string: ${{github.event.pull_request.body}}
+
+    - name: Checkout ThunderTools - ${{steps.tools.outputs.first_match}}
+      if: contains(github.event.pull_request.body, '[DependsOn=ThunderTools:')
+      uses: actions/checkout@v3
+      with:
+        path: ThunderTools
+        repository: rdkcentral/ThunderTools
+        ref: ${{steps.tools.outputs.first_match}}
+
+# ----- Building -----
+    - name: Install necessary packages
+      uses: nick-fields/retry@v2
+      with:
+        timeout_minutes: 10
+        max_attempts: 10
+        command: |
+          sudo gem install apt-spy2
+          sudo apt-spy2 fix --commit --launchpad --country=US
+          sudo apt-get update
+          sudo apt install python3-pip
+          pip install jsonref
+          sudo apt install build-essential cmake ninja-build libusb-1.0-0-dev zlib1g-dev libssl-dev
+
+    - name: Install generators
+      run: |
+        cmake -G Ninja -S ThunderTools -B ${{matrix.build_type}}/build/ThunderTools \
+        -DCMAKE_INSTALL_PREFIX=${{matrix.build_type}}/install/usr
+        cmake --build ${{matrix.build_type}}/build/ThunderTools --target install
+
+    - name: Regex Thunder
+      if: contains(github.event.pull_request.body, '[Options:')
+      id: regexthunder
+      uses: AsasInnab/regex-action@v1
+      with:
+        regex_pattern: '(?<=\[Options:).*(?=\])'
+        regex_flags: 'gim'
+        search_string: ${{github.event.pull_request.body}}
+
+    - name: Build Thunder with archives
+      run: |
+        cmake -G Ninja -S Thunder -B ${{matrix.build_type}}/build/Thunder \
+        -DCMAKE_CXX_FLAGS="-Wall -Wextra -Wpedantic -Werror" \
+        -DBUILD_SHARED_LIBS=OFF \
```

Comment:
What is the point of having exactly the same Action, just with the DBUILD_SHARED_LIBS=OFF?
If we really want to build with both shared libs ON and OFF, we could add 3 additional build types, so that we have 6 and it would be in the same action, so like this:
```
strategy:
  matrix:
    build_type: [Debug, Release, MinSizeRel]
    shared_libs: [ON, OFF]
```

Reply:
To answer your first part: Me having no clue how git workflows work :smiling_face_with_tear: I'll learn making stupid mistakes. I'll change it.

Reply:
Everyday is a school day 😄

If we build with shared libs both on and off, then we have to account for it in Actions in other repos. If we want to build e.g. ThunderInterfaces based on both versions of Thunder (the ones with shared libs ON as well as OFF), then we have to make some adjustments with how artifacts are created and then unpacked (artifact are the way to forwarding output files from one job to the next).

So, before doing that, we have to decide whether we want to build everything with and without shared libs, or not. Thoughts? @bramoosterhuis @pwielders

Reply:
IHMO our components don't need to be tested to be statically linked, maybe this demo repo: 
https://github.com/Metrological/ThunderStaticDemo

Reply:
I think so too, but on the other hand different jobs within the same action are done in paralel, so it is not like we are causing actions to take twice as long with building both versions. Maybe let's hear @pwielders opinion as well. 

On the side note, I don't have access to this demo repo 😄

Reply:
OK i fixed the rights of the repo